feature_engine/creation/binary_combination_feature_reference.py

```diff
@@ -0,0 +1,250 @@
+from typing import List, Optional, Union
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import _find_or_check_numerical_variables
+
+
+class BinaryCombinatioByFeatureReference(BaseEstimator, TransformerMixin):
```

Comment:
class name: `CombineWithFeatureReference`? or `CombineWithReferenceFeature`?
A tiny bit shorter.

I am also thinking if it would be worth expanding to include more operations, like (add, sub, mul, div, mod, pow) . See docs here:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.div.html#pandas.DataFrame.div